sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client_async.py

```diff
@@ -66,25 +68,25 @@ async def close(self):
 
     def _create_pipeline(self, credential, **kwargs):
         # type: (Any, **Any) -> Tuple[Configuration, Pipeline]
-        self._credential_policy = None
+        self._credential_policy = None # type: T
         if hasattr(credential, 'get_token'):
-            self._credential_policy = AsyncBearerTokenCredentialPolicy(credential, STORAGE_OAUTH_SCOPE)
+            self._credential_policy = cast(T, AsyncBearerTokenCredentialPolicy(credential, STORAGE_OAUTH_SCOPE))
         elif isinstance(credential, SharedKeyCredentialPolicy):
-            self._credential_policy = credential
+            self._credential_policy = cast(T, credential)
         elif credential is not None:
             raise TypeError("Unsupported credential: {}".format(credential))
         config = kwargs.get('_configuration') or create_configuration(**kwargs)
         if kwargs.get('_pipeline'):
             return config, kwargs['_pipeline']
-        config.transport = kwargs.get('transport')  # type: ignore
+        _transport = kwargs.get("transport")
```

Comment:
Breaking change, you change the behavior of `config.transport` which is `None` now